Some tests fail with a StackOverflow

Just started looking at this lib as I'd like to integrate it with hxgenjs. Checked it out locally and trying to run tests after compiling with Haxe 4RC2:

```
make                                                                                                                                                   [2.5.0]
haxe test.hxml
node test
Running test case `basic-types.txt`...
Running test case `callback.txt`...
Running test case `ctors.txt`...
Running test case `doc.txt`...
Running test case `eithertype.txt`...
Running test case `empty.txt`...
Running test case `expose.txt`...
Running test case `function.txt`...
Stack overflow
Haxe compilation failed!
Running test case `generic.txt`...
Running test case `readonly.txt`...
Running test case `structs.txt`...
Stack overflow
Haxe compilation failed!
Result: 11. Failed: 2.
Makefile:2: recipe for target 'all' failed
make: *** [all] Error 1
```

Not sure if this affects the actual usage of the lib, will report if so.

--- a/src/hxtsdgen/TypeRenderer.hx
+++ b/src/hxtsdgen/TypeRenderer.hx
@@ -43,6 +43,10 @@ class TypeRenderer {
                     case [{pack: [], name: "Void"}, _]:
                         "void";
 
+                    case [{pack: [], name: "Null"}, [realT]]: // Haxe 4.x
+                        // TODO: generate `| null` union unless it comes from an optional field?
+                        renderType(ctx, realT, paren);
+
                     case [{pack: ["haxe", "extern"], name: "EitherType"}, [aT, bT]]:
                         '${renderType(ctx, aT, true)} | ${renderType(ctx, bT, true)}';
 
@@ -61,7 +65,7 @@ class TypeRenderer {
 
             case TType(_.get() => dt, params):
                 switch [dt, params] {
-                    case [{pack: [], name: "Null"}, [realT]]:
+                    case [{pack: [], name: "Null"}, [realT]]: // Haxe 3.x
                         // TODO: generate `| null` union unless it comes from an optional field?
                         renderType(ctx, realT, paren);
 

